### PR TITLE
Vinegar contribution layer

### PR DESCRIPTION
**Pull request by ralesi**
This simplifies `dired-mode` to behave much like tpope's vinegar modifies `netrw`

### CODE REVIEW

File: contrib/vinegar/packages.el [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+;; Copyright (c) 2014-2015 Sylvain Benner & Contributors
+;;
+;; Author: Sylvain Benner <sylvain.benner@gmail.com>
+;; URL: https://github.com/syl20bnr/spacemacs
+;;
+;; This file is not part of GNU Emacs.
+;;
+;;; License: GPLv3
+
+(defvar vinegar-packages
```
Comment (person808) [inline contrib/vinegar/packages.el]: Should be `setq` now

File: contrib/vinegar/packages.el [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  '(
+    ;; package vinegars go here
+    dired-k
+    dired-ranger
+    dired+
+    )
+  "List of all packages to install and/or initialize. Built-in packages
+which require an initialization must be listed explicitly in the list.")
+
+(defvar vinegar-excluded-packages '()
```
Comment (person808) [inline contrib/vinegar/packages.el]: Same here

**Comment by robbyoconnor** [comment]
One commit and rebase your commit on top of develop...

**Comment by ralesi** [comment]
It was current when I submitted yesterday.  I will do final single commit when everything is in order and we are OK to merge. Until then I figured it would be better to keep separate commits based on feedback to track changes.

**Comment by ralesi** [comment]
Also,  should this be moved to the `! vim` folder?

**Comment by robbyoconnor** [comment]
Might want to also tag it as WIP ;)

**Comment by person808** [comment]
> Also, should this be moved to the !vim folder?

Probably, since it ports a vim plugin.

**Comment by ralesi** [comment]
Updated to !vim and rebased to current version.  Let me know if there is any feedback.

**Comment by syl20bnr** [comment]
@ralesi can you remove the carriage-returns ?

**Comment by ralesi** [comment]
@syl20bnr, I reformatted everything to unix using `C-x RET f`this is not visible to me like it was in Vim when editing files?  I am thinking of adding the the rhs of powerline.  Is `buffer-file-coding-system` the right variable to be looking at.

**Comment by ralesi** [comment]
I paired this contrib down a bit.  It now concentrates on only minimizing the dired buffer.

**Comment by bcarrell** [comment]
Any chance of this being merged?  I'd like to stop maintaining my version.

**Comment by ralesi** [comment]
@bcarrell, I will look into it, there are a few ways details can be reduced, and I will make sure symlinks are still showing.

**Comment by syl20bnr** [comment]
Never too late to merge :-)
Thank you ! :+1: 
Cherry-picked into `develop` branch, you can safely delete your branch.

**Comment by sooheon** [comment]
Great layer, @ralesi. Just curious what your rationale is for having `revert-buffer` on C-r and `dired-do-redisplay` on r? Redisplaying only the selected file seems to be a more niche command than redisplaying the buffer, no?

**Comment by ralesi** [comment]
@soohoen, I will change it,  I will make `r` do both,  will that work?

**Comment by sooheon** [comment]
Simply having `revert-buffer` on r would take care all use cases I can think of. I don't see why if you want to redisplay some items, you wouldn't go ahead and refresh everything, seeing as it takes no time at all.

**Comment by sooheon** [comment]
@ralesi on the latest develop, this layer's rebinding of keys does not work anymore. I guess it has something to do with evilification changes, do you mind looking into it once more?

**Pull request closed without merging**